packages/commercetools/api-client/src/helpers/cart/actions.ts

```diff
@@ -22,44 +22,44 @@ export const createChangeLineItemQuantityAction = (product: LineItem) => ({
   }
 });
 
-export const setShippingAddressAction = (shippingDetails: AddressInput) => ({
+export const setShippingAddressAction = (shippingDetails: Address) => ({
   setShippingAddress: {
     address: {
       firstName: shippingDetails.firstName,
       lastName: shippingDetails.lastName,
       streetName: shippingDetails.streetName,
-      streetNumber: '',
+      streetNumber: shippingDetails.streetNumber,
       city: shippingDetails.city,
-      // TODO:: selecting state
       // state: shippingDetails.state,
       postalCode: shippingDetails.postalCode,
       country: shippingDetails.country,
-      phone: shippingDetails.phone
+      phone: shippingDetails?.contactInfo?.phone
     }
   }
 });
 
-export const setShippingMethodAction = (shippingMethodId: string) => ({
+export const setShippingMethodAction = (shippingMethodId?: string) => ({
   setShippingMethod: {
-    shippingMethod: {
-      id: shippingMethodId
-    }
+    shippingMethod: shippingMethodId ? { id: shippingMethodId } : null
   }
 });
 
-export const setBillingAddressAction = (shippingDetails: AddressInput) => ({
+export const addPayment = (payment: ResourceIdentifierInput) => ({
+  addPayment: { payment }
+});
+
+export const setBillingAddressAction = (billingDetails: Address) => ({
   setBillingAddress: {
     address: {
-      firstName: shippingDetails.firstName,
-      lastName: shippingDetails.lastName,
-      streetName: shippingDetails.streetName,
-      streetNumber: '',
-      city: shippingDetails.city,
-      // TODO:: selecting state
-      // state: shippingDetails.state,
-      postalCode: shippingDetails.postalCode,
-      country: shippingDetails.country,
-      phone: shippingDetails.phone
+      firstName: billingDetails.firstName,
+      lastName: billingDetails.lastName,
+      streetName: billingDetails.streetName,
+      streetNumber: billingDetails.streetNumber,
+      city: billingDetails.city,
+      // state: billingDetails.state,
+      postalCode: billingDetails.postalCode,
+      country: billingDetails.country,
+      phone: billingDetails.contactInfo.phone
```

Comment:
if think that it can fail here https://github.com/DivanteLtd/vue-storefront/pull/4348/files#diff-0aba958830f4b68174ff6ab8ce9e6e2bR36 then probably you should change it to `billingDetails?.contactInfo?.phone`